pageserver/compaction/tests/tests.rs

```diff
@@ -0,0 +1,37 @@
+use pageserver_compaction::interface::CompactionLayer;
+use pageserver_compaction::simulator::MockTimeline;
+
+/// Test the extreme case that there are so many updates for a single key that
+/// even if we produce an extremely narrow delta layer, spanning just that one
+/// key, we still too many records to fit in the target file size. We need to
+/// split in the LSN dimension too in that case.
+///
+/// TODO: The code to avoid this problem has not been implemented yet! So the
```

Comment:
You probably want to merge this with the TODO unaddressed, but when do we want to address this? Before or after we activate the new compaction?

Reply:
Yes, this needs to be addressed before we can switch to the new compaction. Otherwise, we can end up with a layer file that's too large to be uploaded to S3 without multipart upload, so the upload will fail, and the timeline will be stuck trying to upload it. We had that problem before...

Reply:
Semi-related: I must've accidentially removed the size check with the new layer impl as I cannot see this PR removing it either.